src/sentry/event_manager.py

```diff
@@ -1820,24 +1797,27 @@ def handle_existing_grouphash(
     grouphashes to the group.
     """
 
-    # There is a race condition here where two processes could "steal"
-    # hashes from each other. In practice this should not be user-visible
-    # as group creation is synchronized, meaning the only way hashes could
-    # jump between groups is if there were two processes that:
-    #
-    # 1) have BOTH found an existing group
-    #    (otherwise at least one of them would be in the group creation
-    #    codepath which has transaction isolation/acquires row locks)
-    # 2) AND are looking at the same set, or an overlapping set of hashes
-    #    (otherwise they would not operate on the same rows)
-    # 3) yet somehow also retrieve different groups here
-    #    (otherwise the update would not change anything)
-    #
-    # We think this is a very unlikely situation. A previous version of
-    # _save_aggregate had races around group creation which made this race
-    # more user visible. For more context, see 84c6f75a and d0e22787, as
-    # well as GH-5085.
-    group = Group.objects.get(id=existing_grouphash.group_id)
+    if isinstance(existing_group_or_grouphash, GroupHash):
```

Comment:
why not lift this logic to callers and instead simply have this method use `Group`? could be a bit more straightforward

Reply:
I could. Three of the four times it's used, it's used on grouphashes rather than groups, though. Perhaps better would be to change the one spot where it gets passed a group to pass it a grouphash instead?

Reply:
yeah that would also make sense! but i'll defer to you on how you'd like to implement

Reply:
I ended up going that route. I've updated the PR description.